baybe/objectives/base.py

```diff
@@ -0,0 +1,47 @@
+"""Base classes for all objectives."""
+
+from abc import ABC, abstractmethod
+from typing import Union
+
+import pandas as pd
+from attrs import define
+
+from baybe.objectives.deprecation import structure_objective
+from baybe.serialization.core import (
+    converter,
+    unstructure_base,
+)
+from baybe.serialization.mixin import SerialMixin
+from baybe.targets.base import Target
+
+
+@define(frozen=True)
+class Objective(ABC, SerialMixin):
+    """Abstract base class for all objectives."""
+
+    @property
+    @abstractmethod
+    def targets(self) -> tuple[Target, ...]:
```

Comment:
Just to make sure: Internally, we still use a tuple of targets, but the code was changed in such a way that a user can also just provide a single target without running into any trouble, right?

Reply:
Yes, or rather: if you use `SingleTargetObjective`, you *must* provide exactly one target (not in a list/collection but directly) while `DesirabiltyObjective` specifically asks for a list of at least two target. If OK -> resolve 🙃